scripts/envsetup.sh

```diff
@@ -35,23 +39,21 @@ fi
 METADIR=${METADIR:-${SOURCEDIR}/../..}
 
 if [[ ! -f "${BUILDDIR}/conf/local.conf" ]]; then
-  declare -a DISTRO_CONFIGS=${supported_distros[$DISTRO]}
-  [[ -n ${DISTRO_CONFIGS[@]} ]] && { echo "Using (${DISTRO_CONFIGS[*]}) for the specified distro '$DISTRO'"; } || { echo "The specified distro $DISTRO is not supported"; return 1; }
-
   source "$METADIR/poky/oe-init-build-env" "$BUILDDIR"
 
   echo "METADIR  := \"\${@os.path.abspath('${METADIR}')}\"" >> conf/bblayers.conf
   cat "${METADIR}/meta-updater/conf/include/bblayers/sota.inc" >> conf/bblayers.conf
   cat "${METADIR}/meta-updater/conf/include/bblayers/sota_${MACHINE}.inc" >> conf/bblayers.conf
+
   sed -e "s/##MACHINE##/$MACHINE/g" \
       -e "s/##DISTRO##/$DISTRO/g" \
          "${METADIR}/meta-updater/conf/$BASE_CONF" >> conf/local.conf
 
-  for config in ${DISTRO_CONFIGS[@]}; do
+  for config in ${supported_distros[$DISTRO]}; do
     if [[ "$BASE_CONF" != "$config" ]]; then
       cat "${METADIR}/meta-updater/conf/$config" >> conf/local.conf
```

Comment:
I suppose this `cat` will fail if a user specifies unsupported distro, although printing clear described failure reason and returning might be a bit better.

Reply:
Good point